apps/mymedicare_cb/models.py

```diff
@@ -185,145 +173,63 @@ def get_and_update_user(
-        })
-        logger.info(log_dict)
-        raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-
-    if user_hicn_hash is None:
-        mesg = "user_hicn_hash can not be None"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-    else:
-        if len(user_hicn_hash) != 64:
-            mesg = "incorrect user HICN hash format"
-            log_dict.update({
-                "status": "FAIL",
-                "mesg": mesg,
-            })
-            logger.info(log_dict)
-            raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-
-    # If mbi_hash is not NULL, perform length check.
-    if user_mbi_hash is not None:
-        if len(user_mbi_hash) != 64:
-            mesg = "incorrect user MBI hash format"
-            log_dict.update({
-                "status": "FAIL",
-                "mesg": mesg,
-            })
-            logger.info(log_dict)
-            raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-
-    if fhir_id is None:
-        mesg = "fhir_id can not be None"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-
-    if fhir_id == "":
-        mesg = "fhir_id can not be an empty string"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise BBMyMedicareCallbackCrosswalkCreateException(mesg)
-
-    if User.objects.filter(username=username).exists():
-        mesg = "user already exists"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise ValidationError(mesg, username)
-
-    if Crosswalk.objects.filter(_user_id_hash=user_hicn_hash).exists():
-        mesg = "user_hicn_hash already exists"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise ValidationError(mesg, user_hicn_hash)
-
-    # If mbi_hash is not NULL, perform check for duplicate
-    if user_mbi_hash is not None:
-        if Crosswalk.objects.filter(_user_mbi_hash=user_mbi_hash).exists():
-            mesg = "user_mbi_hash already exists"
-            log_dict.update({
-                "status": "FAIL",
-                "mesg": mesg,
-            })
-            logger.info(log_dict)
-            raise ValidationError(mesg, user_hicn_hash)
-
-    if fhir_id and Crosswalk.objects.filter(_fhir_id=fhir_id).exists():
-        mesg = "fhir_id already exists"
-        log_dict.update({
-            "status": "FAIL",
-            "mesg": mesg,
-        })
-        logger.info(log_dict)
-        raise ValidationError(mesg, fhir_id)
+    _validate_asserts(logger, log_dict, [
```

Comment:
This is a lot cleaner than what was there before - where did you get this pattern from?